app/code/Magento/Reports/Test/Unit/Model/ResourceModel/Product/CollectionTest.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Reports\Test\Unit\Model\ResourceModel\Product;
+
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+
+/**
+ * Test for Magento\Reports\Model\ResourceModel\Product\Collection.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```

Comment:
While coupling is unavoidable you cannot suppress `ExcessiveMethodLength` in new code, please refactor test accordingly.

Use imports actively (Alt+Enter in PhpStorm) to fit method calls in one line, `getMockForAbstractClass` for interface could be replaced with `createMock`.

After that please squash changes into single commit and force push into the same branch.